### PR TITLE
 PMM-4249 Add API data retention.

### PR DESCRIPTION
**Pull request by askomorokhov**
https://jira.percona.com/browse/PMM-4249

Blocks https://github.com/percona/pmm-managed/pull/245

FB: Percona-Lab/pmm-submodules#383

### CODE REVIEW

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-409/

File: api/serverpb/server.proto [hunk 1]
```diff
@@ -109,11 +109,18 @@ message MetricsResolutions {
   google.protobuf.Duration lr = 3;
 }
 
+// QAN contains query analytics configuration.
+message QAN {
+  // Number of days of data retain.
+  uint32 data_retention = 1;
```
Comment (AlekSi) [inline api/serverpb/server.proto]: It should be `google.protobuf.Duration`.

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-409/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-409/

File: api/serverpb/server.proto [hunk 2]
```diff
@@ -111,11 +111,18 @@ message MetricsResolutions {
   google.protobuf.Duration lr = 3;
 }
 
+// QAN contains query analytics configuration.
+message QAN {
+  // Number of days of data retaintion in format: '24h'.
```
Comment (AlekSi) [inline api/serverpb/server.proto]: FYI, https://developers.google.com/protocol-buffers/docs/proto3#json:
> Generated output always contains 0, 3, 6, or 9 fractional digits, depending on required precision, **followed by the suffix "s"**. Accepted are any fractional digits (also none) as long as they fit into nano-seconds precision and **the suffix "s" is required**.

We can't change that on the API level, but we will make it nicer in CLI and UI.

I will update that comment.

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-409/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-409/

**Pull request merged**